packages/core-mobile/ios/AppDelegate.mm

```diff
@@ -78,4 +78,10 @@ - (void)loadRocketSimConnect {
 #endif
 }
 
+- (void)applicationWillEnterForeground:(UIApplication *)application {
+  // this only reset the badgeCount in app group userDefaults,
+  // the actual badge count is reset in handleNotificationCleanup
+  // inside react native project
+    [[[NSUserDefaults alloc] initWithSuiteName:@"group.org.avalabs.corewallet"] setInteger:0 forKey:@"badgeCount"];
```

Comment:
what does this mean? let's say we don't call the react native logic to reset, what will happen in iOS? also how is Android behaving?

Reply:
since we don't really access shared application in extension, so we rely on userDefault to increment the count.  and the react native logic handles reset for both iOS and Android.